clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/cache/AzureServerGroupCachingAgent.groovy

```diff
@@ -255,69 +135,114 @@ class AzureServerGroupCachingAgent extends AzureCachingAgent {
     }
 
     def cacheResult = metricsSupport.transformData {
-      if (updatedServerGroup) {
-        return buildCacheResult(providerCache, null, 0, updatedServerGroup, null)
-      } else {
-        def appName = AzureUtilities.getAppNameFromAzureResourceName(serverGroupName)
-        evictedServerGroup = new AzureServerGroupDescription(
-          name: serverGroupName,
-          region: region,
-          appName: appName,
-          cloudProvider: "azure",
-          clusterName: appName,
-          lastReadTime: System.currentTimeMillis()
+      buildCacheResult([newServerGroup], [:], [])
+    }
+
+    if (cacheResult.cacheResults.values().flatten().isEmpty()) {
+      // evict it
+      providerCache.evictDeletedItems(AZURE_ON_DEMAND.ns, [serverGroupKey])
```

Comment:
What if a cache aget is in the middle of the execution when the "delete" OnDemand event happens? It will basically restore back the "old" server group, right?